### PR TITLE
Update supported collectors doc to organize by type

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

This PR dramatically changes our supported collectors list document so collectors are organized by type, not orchestrator.

This builds off the work @ilyam8 did a few months back to help me put all the collectors into a single document, but since then it's become clearer that we should organize collectors by their service, not what orchestrator they belong to.

##### Component Name

area/docs
area/collectors

### CODE REVIEW

File: collectors/COLLECTORS.md [hunk 1]
```diff
@@ -1,219 +1,478 @@
...
+
+### Logs
+
+-   [Fluentd](https://learn.netdata.cloud/docs/agent/collectors/go.d.plugin/modules/fluentd/): Gather application
+    plugins metrics from an endpoint provided by `in_monitor plugin`.
+-   [Logstash](https://learn.netdata.cloud/docs/agent/collectors/go.d.plugin/modules/logstash/): Monitor JVM threads,
+    memory usage, garbage collection statistics, and more.
+-   [OpenVPN status logs](/collectors/python.d.plugin/ovpn_status_log/): Parse server log files and provide summary
+    (client, traffic) metrics.
+-   [Web server logs (Go version for Apache,
```
Comment (ilyam8) [inline collectors/COLLECTORS.md:155]: we have separate go collector for squid logs

> https://github.com/netdata/go.d.plugin/tree/master/modules/squidlog

File: collectors/COLLECTORS.md [hunk 2]
```diff
@@ -1,219 +1,478 @@
...
+-   [Pulsar](https://learn.netdata.cloud/docs/agent/collectors/go.d.plugin/modules/pulsar/): Collect summary,
+    namespaces, and topics performance statistics.
+-   [RabbitMQ (Go)](https://learn.netdata.cloud/docs/agent/collectors/go.d.plugin/modules/rabbitmq/): Collect message
+    broker overview, system and per virtual host metrics.
+-   [RabbitMQ (Python)](/collectors/python.d.plugin/rabbitmq/README.md): Collect message broker global and per virtual
+    host metrics. 
+-   [VerneMQ](https://learn.netdata.cloud/docs/agent/collectors/go.d.plugin/modules/vernemq/): Monitor MQTT broker
+    health and performance metrics. It collects all available info for both MQTTv3 and v5 communication
+
+### Network
```
Comment (ilyam8) [inline collectors/COLLECTORS.md:176]: What does `network` means? i see it contains:

- dhcp
- dns
- ntp
- vpn (openvpn, libreswan)
- aaa (freeradius)
- voice/video (openSIPS)

File: collectors/COLLECTORS.md (hunk 1 above)
Comment (joelhans) [reply collectors/COLLECTORS.md:155]: Added.

File: collectors/COLLECTORS.md (hunk 2 above)
Comment (joelhans) [reply collectors/COLLECTORS.md:178]: It's kind of a catchall category for anything to do with networks and/or sending data across networks, and doesn't fit better elsewhere. I'm trying to minimize the number of categories/types as much as possible.

Comment (ilyam8) [reply collectors/COLLECTORS.md:178]: > a catchall category
> sending data across networks
> trying to minimize the number of categories/types as much as possible.

Yeah, i agree, it is either you have a lot of categories or have _a catchall category_.

**Comment by joelhans** [comment]
@ilyam8 Any other thoughts on organizing collectors this way? I also just added Prometheus to the list under a new **generic** heading.

Speaking of Prometheus, we may want to consider adding new pages for some of the new things we can collect this way, if the service is popular enough and we couldn't do it previously.

**Comment by ilyam8** [comment]
> Any other thoughts on organizing collectors this way?

well, i think it is ~ ok.

**Review by amoss — APPROVED** [review]
New structure is much easier to read. Can't find any copy problems in the text.

**Comment by joelhans** [comment]
Just adding a few more reviewers in hopes of pushing this change.

File: collectors/COLLECTORS.md [hunk 3]
```diff
@@ -1,219 +1,486 @@
...
-real-time, interactive charts. The following list includes collectors for both internal system metrics, and external
-apps/services metrics.
+real-time, interactive charts. The following list includes collectors for both external services/applications and
+internal system metrics.
 
 Read more about collectors and how to enable them in our [collectors documentation](/collectors/README.md), or use the
 [collector quickstart](/collectors/QUICKSTART.md) to figure out how to collect metrics from your favorite app/service
 with auto-detection and minimal configuration.
 
+Some collectors have both Go and Python versions. While the Go verisons are newer, they are most often disabled by
```
Comment (aabatangle) [inline collectors/COLLECTORS.md:17]: verisons --> versions

Comment (joelhans) [reply collectors/COLLECTORS.md]: Fixed this.

**Review by amoss — APPROVED** [review]
Reapproving

**Review by aabatangle — APPROVED** [review]

**Pull request merged**